### PR TITLE
Added Graph to SPARQL resolve query to avoid doble labels

### PR DESCRIPTION
**Pull request by crisely09**
Addresses this issue:
https://github.com/BlueBrain/nexus-forge/issues/309

See example notebook.

### CODE REVIEW

**Comment by crisely09** [comment]
@MFSY, I think we need to add:
```with:
  token: ${{ secrets.CODECOV_TOKEN }}
```
to the test configuration, if we want the codecov to stop failing

File: kgforge/specializations/resolvers/ontology_resolver.py [hunk 1]
```diff
@@ -57,33 +60,35 @@ def _resolve(self, text: Union[str, List[str]], target: Optional[str], type: Opt
...
-            ?id notation ?notation .
-            }}    
-            {{
-            SELECT * WHERE {{
-                {{ {0} ; label ?label {1} }} UNION
-                {{ {0} ; notation ?notation {2} }} UNION
-                {{ {0} ; prefLabel ?prefLabel {3} }} UNION
-                {{ {0} ; altLabel ?altLabel {4} }}
-            }} LIMIT {5}
+            GRAPH ?g {{
```
Comment (MFSY) [inline kgforge/specializations/resolvers/ontology_resolver.py:63]: I would add Graph clause in the [agent resolver](https://github.com/BlueBrain/nexus-forge/blob/f94700222e78968e00849aa07d7e978f5109bd89/kgforge/specializations/resolvers/agent_resolver.py#L53) as well

File: kgforge/specializations/stores/nexus/service.py [hunk 2]
```diff
@@ -265,6 +266,9 @@ def resolve_context(self, iri: str, local_only: Optional[bool] = False) -> Dict:
             else:
                 raise ValueError(f"{context_to_resolve} is not resolvable")
         else:
+            # Make sure context is not deprecated
+            if resource['_deprecated']:
```
Comment (MFSY) [inline kgforge/specializations/stores/nexus/service.py:270]: I would check _deprecated is in the dict first.

**Comment by MFSY** [comment]
>   token: ${{ secrets.CODECOV_TOKEN }}

How can we get a token ?

File: kgforge/specializations/stores/nexus/service.py (hunk 2 above)
Comment (crisely09) [reply kgforge/specializations/stores/nexus/service.py:270]: is there a case where it is not?

**Comment by crisely09** [comment]
> 
I would expect is the same as for version 2:

https://about.codecov.io/blog/getting-started-with-the-codecov-api-v2/

File: kgforge/specializations/resolvers/ontology_resolver.py (hunk 1 above)
Comment (crisely09) [reply kgforge/specializations/resolvers/ontology_resolver.py:63]: done

**Comment by MFSY** [comment]
Okay. Added in #311 a CODECOV_TOKEN as secret and used it in test github action. Let see if it works

File: kgforge/specializations/resolvers/ontology_resolver.py [hunk 3]
```diff
@@ -39,12 +39,15 @@ def mapper(self) -> Callable:
         return DictionaryMapper
 
     def _resolve(self, text: Union[str, List[str]], target: Optional[str], type: Optional[str],
-                 strategy: ResolvingStrategy, resolving_context: Any, limit: Optional[str], threshold=Optional[float]) \
+                 strategy: ResolvingStrategy, resolving_context: Any, limit: Optional[int], threshold=Optional[float]) \
```
Comment (MFSY) [inline kgforge/specializations/resolvers/ontology_resolver.py:42]: The limit: Optional[int] should also be done in all [resolvers](https://github.com/BlueBrain/nexus-forge/tree/f1bdac4a737b6936b5fd3425222c3db0459de206/kgforge/specializations/resolvers)

**Review by MFSY — APPROVED** [review]

**Pull request merged**